src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -136,11 +146,19 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     configuration.setBoolean("fs.gs.impl.disable.cache", true);
     configuration.setBoolean("fs.gs.metadata.cache.enable", false);
 
+    String query = generateQuery();
+
+    if (query != null) {
+      configuration.set(BigQueryConfiguration.INPUT_QUERY_KEY, query);
+    }
+
     String temporaryGcsPath = String.format("gs://%s/hadoop/input/%s", bucket, uuid);
-    AvroBigQueryInputFormat.setTemporaryCloudStorageDirectory(configuration, temporaryGcsPath);
-    AvroBigQueryInputFormat.setEnableShardedExport(configuration, false);
+    CdapBigQueryInputFormat.setTemporaryCloudStorageDirectory(configuration, temporaryGcsPath);
```

Comment:
Cdap isn't really a descriptive name. PartitionedBigQueryInputFormat would be better.

Reply:
Done.